org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/controls/ZoneChart.java

```diff
@@ -78,6 +78,54 @@ private CustomChartComposite(final Composite parent, final JFreeChart chart)
 
     private Stack<Rectangle> zoomHistory = new Stack<Rectangle>();
 
+    void fitToData()
+    {
+      final List<Rectangle> zoomRedoHistory = new ArrayList<>(zoomHistory);
+
+      AbstractOperation addOp = new AbstractOperation("Show all data")
+      {
+
+        @Override
+        public IStatus execute(IProgressMonitor monitor, IAdaptable info)
+            throws ExecutionException
+        {
+
+          zoomHistory.clear();
+
+          CustomChartComposite.super.restoreAutoBounds();
+          return Status.OK_STATUS;
+        }
+
+        @Override
+        public IStatus redo(IProgressMonitor monitor, IAdaptable info)
+            throws ExecutionException
+        {
+          zoomHistory.clear();
+
+          CustomChartComposite.super.restoreAutoBounds();
+          return Status.OK_STATUS;
+        }
+
+        @Override
+        public IStatus undo(IProgressMonitor monitor, IAdaptable info)
+            throws ExecutionException
+        {
+
+          CustomChartComposite.super.restoreAutoBounds();
+          zoomHistory.addAll(zoomRedoHistory);
+
+          for (Rectangle h : zoomHistory)
+          {
```

Comment:
Ok, I've had a play. You're storing the Rectangle of the selection (in pixels). But, if the graph/view is resized, those pixels will now point to another region of the chart.

We need to store the data coords of the operation for a Zoom, and for the "Fit to Window". There's a comment in the PR conversation that shows how to get the data coords.  I suggest we could benefit from a utility function that returns the current viewport as a Rectangle2D, oh, and another that sets the viewport to a Rectangle2D.